### PR TITLE
Add body force estimation skeleton

### PR DESCRIPTION
**Pull request by justinyim**
Created body force estimator package skeleton and corresponding body force estimate message.

### CODE REVIEW

File: spirit_msgs/msg/BodyForceEstimate.msg [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+geometry_msgs/Wrench body_wrench
+geometry_msgs/Wrench flu_wrench
+geometry_msgs/Wrench blu_wrench
+geometry_msgs/Wrench fru_wrench
+geometry_msgs/Wrench bru_wrench
+geometry_msgs/Wrench fll_wrench
+geometry_msgs/Wrench bll_wrench
+geometry_msgs/Wrench frl_wrench
+geometry_msgs/Wrench brl_wrench
```
Comment (jcnorby) [inline spirit_msgs/msg/BodyForceEstimate.msg:16]: Would we rather this be an array, i.e. geometry_msgs/Wrench[] wrenches? Depends on how you want to use this message.

File: body_force_estimator/src/body_force_estimator_node.cpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+int main(int argc, char** argv) {
+	ros::init(argc, argv, "body_force_estimator_node");
+	ros::NodeHandle nh;
+
+	BodyForceEstimator body_force_estimator(nh);
+	body_force_estimator.spin();
+
+	return 0;
+}
```
Comment (jcnorby) [inline body_force_estimator/src/body_force_estimator_node.cpp:12]: Fix indentation (should be 2 spaces)

File: body_force_estimator/test/test_body_force_estimator.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	BodyContactEstimator body_force_estimator(nh);
+	EXPECT_EQ(1 + 1, 2);
+}
+
+int main(int argc, char** argv) {
+	testing::InitGoogleTest(&argc, argv);
+	ros::init(argc, argv, "body_force_estimator_tester");
+
+	return RUN_ALL_TESTS();
+}
```
Comment (jcnorby) [inline body_force_estimator/test/test_body_force_estimator.cpp:17]: Fix indentation (should be 2 spaces)

File: body_force_estimator/src/body_force_estimator_node.cpp (hunk 2 above)
Comment (jcnorby) [reply body_force_estimator/src/body_force_estimator_node.cpp:12]: (We also need to fix it in other places but might as well maintain the standard until we do)

**Comment by jcnorby** [comment]
Added some pretty minor comments, once they're resolved I'll go ahead and approve

File: body_force_estimator/src/body_force_estimator.cpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#include "body_force_estimator/body_force_estimator.h"
+
+BodyForceEstimator::BodyForceEstimator(ros::NodeHandle nh) {
+  nh_ = nh;
+
+  // Load rosparams from parameter server
+  std::string state_estimate_topic, body_force_topic;
+  nh.param<std::string>("topics/state_estimate", state_estimate_topic, "/state_estimate");
+  nh.param<std::string>("topics/body_force", body_force_topic, "/body_force");
+  nh.param<double>("body_force_estimator/update_rate", update_rate_, 200); // add a param for your package instead of using the estimator one
```
[the remark this answers is not in the document]
Comment (justinyim) [reply body_force_estimator/src/body_force_estimator.cpp:10]: Added

File: spirit_msgs/msg/BodyForceEstimate.msg (hunk 1 above)
Comment (justinyim) [reply spirit_msgs/msg/BodyForceEstimate.msg]: Yes, an array does seem nicer.

File: body_force_estimator/src/body_force_estimator_node.cpp (hunk 2 above)
Comment (justinyim) [reply body_force_estimator/src/body_force_estimator_node.cpp]: Indentation fixed in this and other files.

**Review by jcnorby — APPROVED** [review]

**Comment by jcnorby** [comment]
Looks good from my end, although Travis didn't like something. Once travis okays it we can pull it in

**Pull request merged**